### DISCUSSION TITLE
Is it possible to call ulauncher-toggle with initial text?

### DISCUSSION THREAD

**Discussion by ajboni** [Ulauncher/Ulauncher#1036] [Q&A] [answered]
Hello!!
Something like: `ulauncher-toggle  sys`
usecase: I would love to have this as a button in waybar

**Comment by friday** [comment, marked as answer]
It's not possible. It would be nice though. See #312

**Reply by ajboni** [reply]
Ah what a shame, would a PR for this be welcome?  If so, could it be a 'good first issue' ?

**Reply by friday** [reply]
> Ah what a shame, would a PR for this be welcome? If so, could it be a 'good first issue' ?

That would be appreciated :+1: It's not a complex thing to add.

As mentioned [here](https://github.com/Ulauncher/Ulauncher/issues/312#issuecomment-450501179) keywords can be changed, which creates some potential issues with just setting the query directly. I think I was overthinking it and it's best to ignore that however, and think of it as an undocumented "bonus feature" for power users.

If you want to make this simple, add it to `ulauncher-toggle` and send it over by tweaking the current `gdbus` command, which communicates with the application as defined in main.py (for historic reasons). Maybe you need to wrap "Activate" with a custom method or send two dbus calls.

Make sure it doesn't break the behavior for users who enabled the setting to continuing from the previous session and doesn't pass any arguments to `ulauncher-toggle`.

**Reply by ajboni** [reply]
Thanks for the summary! I will give this a try this week.
I like the idea of being able to use extension_id AND still have the possibility to directly inputting  the  query
`ulauncher-toggle --query sys` and `ulauncher-toggle --query com.github.tjquillan.ulauncher-system/system_ext` .
But will the user need to know the keyword_id for the extension? as defined in `manifest.json`?

**Reply by friday** [reply]
> I like the idea of being able to use extension_id AND still have the possibility to directly inputting the query
> But will the user need to know the keyword_id for the extension? as defined in manifest.json?

The user would both have to know that and he extension id. This is part of why I no longer think that's worth the effort and it's better to just set the query as text and limit it to that. So I disagree with my old 2018 comment, but just wanted to point out the quirks as I detailed them then, as they are still true.
